### PR TITLE
Profile token3

### PR DESCRIPTION
**Pull request by LasseKallio**
profile changes

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    'tokenString': 'milkandtoastandhoney',
```
Comment (ElenaSimilietova) [inline config.js]: Why did you added this file into repository? it contains secret data and shouldn't be in control version, that's why we moved it to separate file.

File: app/pages/profile/ProfileModule.js [hunk 2]
```diff
@@ -9,6 +9,55 @@ angular.module('profileModule', ['ngRoute'])
...
+          sessionStorage.setItem('dueDateOk', 1);
+       }
+       else
+       {
+          $scope.timestamp = response.data.due_date;
+          $scope.subscriptionInfo = "Your subscription is valid untill ";
+          angular.element(document.getElementById('payMonthButton'))[0].disabled = true;
+          angular.element(document.getElementById('payYearButton'))[0].disabled = true;
+          sessionStorage.setItem('dueDateOk', 0);
+       }
```
Comment (ElenaSimilietova) [inline app/pages/profile/ProfileModule.js:36]: Again, every time user visits profile page, you set dueDateOk. You should do it just once, after log in.

File: app/pages/profile/ProfileModule.js [hunk 3]
```diff
@@ -9,6 +9,55 @@ angular.module('profileModule', ['ngRoute'])
...
+  };   
+	 
+ 	User.subscribe(token, subscribeDueDate).then(function(response) {
+      angular.element(document.getElementById('payMonthButton'))[0].disabled = true;
+      angular.element(document.getElementById('payYearButton'))[0].disabled = true;
+      $scope.subscriptionInfo = "Thanks for your subscription!";
+      sessionStorage.setItem('dueDateOk', 0);
+   		$location.path('/users/profile');
+  	});
+
```
Comment (ElenaSimilietova) [inline app/pages/profile/ProfileModule.js]: indents

File: app/pages/profile/profile.html [hunk 4]
```diff
@@ -1 +1,19 @@
...
+			<div class="col-md-10 profile-general"">
+			<h2>Hello, {{firstName}}</h2>
+			<br>
+			{{subscriptionInfo}}{{timestamp | date }}	
+			</br></br>
+			<button class="btn btn-primary payMonthButton"  id ="payMonthButton" ng-click="submitPayment(1)">Pay for a month</button>
+			<button class="btn btn-primary payYearButton" id ="payYearButton" ng-click="submitPayment(12)">Pay for a year</button>
+			<div data-ng-init="getPayment()">
+    		</div>
+			</div>
```
Comment (ElenaSimilietova) [inline app/pages/profile/profile.html]: indents

File: backend/users.js [hunk 5]
```diff
@@ -132,3 +132,68 @@ exports.logOut = function(req, res) {
...
+          connection.release();
+          res.status(500).send({message: 'DB connection error'});
+        }
+        connection.query("UPDATE users SET due_date = CURRENT_TIMESTAMP + INTERVAL '" + subscribeDueDate + "' MONTH WHERE ID = '" + user.userID + "'"
+        , function (err, rows) {
+
+          connection.release();
+          if (err) {
+            res.status(500).send({ message: 'Query error'});
+          }
```
Comment (ElenaSimilietova) [inline backend/users.js]: Don't send messages. User should not know the reason of problem. Just catch at frontend status 500 in rejection and show the same message, that my scripts show.

**Comment by LasseKallio** [comment]
Where i put this, becouse it's working with this in

  $scope.submitPayment = function(value) { ....

  sessionStorage.setItem('dueDateOk', 0);



ElenaSimilietova kirjoitti 29.8.2017 21:26:

**Comment by ElenaSimilietova** [comment]
you should put this in SignInController, after
sessionStorage.setItem('token', response.data.token);
sessionStorage.setItem('expiresIn', response.data.expiresIn);

**Pull request merged**